src/Rector/StaticCall/RouteActionCallableRector.php

```diff
@@ -0,0 +1,184 @@
+            new ConfiguredCodeSample(
+                <<<'CODE_SAMPLE'
+Route::get('/users', 'UserController@index');
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+Route::get('/users', [\App\Http\Controllers\UserController::class, 'index']);
+CODE_SAMPLE
+            ,
+                [
+                    self::NAMESPACE => 'App\Http\Controllers',
+                ]
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param Node\Expr\MethodCall|StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isRouteAdditionMethod($node)) {
+            return null;
+        }
+
+        $position = $this->getActionPosition($node->name);
+
+        if (! isset($node->args[$position])) {
+            return null;
+        }
+
+        $arg = $node->args[$position];
+
+        $action = $this->valueResolver->getValue($arg->value);
+        if (! $this->isActionString($action)) {
+            return null;
+        }
+
+        $segments = explode('@', $action);
+        if (count($segments) !== 2) {
+            return null;
+        }
+
+        [$controller, $method] = $segments;
+        $namespace = $this->getNamespace($this->file->getSmartFileInfo());
+        if (! str_starts_with($controller, '\\')) {
+            $controller = $namespace . '\\' . $controller;
+        }
+
+        $scope = $node->getAttribute(AttributeKey::SCOPE);
+
+        $phpMethodReflection = $this->reflectionResolver->resolveMethodReflection($controller, $method, $scope);
+
+        if (! $phpMethodReflection instanceof PhpMethodReflection) {
+            return null;
+        }
+
+        $node->args[$position]->value = $this->nodeFactory->createArray([
+            $this->nodeFactory->createClassConstReference($controller),
+            $method,
+        ]);
+        return $node;
+    }
+
+    public function configure(array $configuration): void
+    {
+        $this->routes = $configuration[self::ROUTES] ?? [];
+        if (isset($configuration[self::NAMESPACE])) {
+            $this->namespace = $configuration[self::NAMESPACE];
+        }
+    }
+
+    private function isRouteAdditionMethod(MethodCall|StaticCall $node): bool
+    {
+        if (! $this->isNames(
+            $node->name,
+            ['any', 'delete', 'get', 'options', 'patch', 'post', 'put', 'match', 'fallback']
+        )) {
+            return false;
+        }
+
+        if ($node instanceof MethodCall && $this->nodeTypeResolver->isObjectTypes(
+            $node->var,
+            [new ObjectType('Illuminate\Routing\Router'), new ObjectType('Illuminate\Routing\RouteRegistrar')]
+        )) {
+            return true;
+        }
+
+        return $node instanceof StaticCall && $this->isName($node->class, 'Illuminate\Support\Facades\Route');
+    }
```

Comment:
This looks like complex re-usable logic. Could you extact is to own service?

Reply:
Extract it to `RouterRegisterNodeAnalyzer`.